### PR TITLE
Allows Ghosts to click Portals and engaged teleporter HUBs to use them

### PR DESCRIPTION
**Pull request by DeityLink**
:cl:
* bugfix: Ghosts can once again click Portals and engaged teleporter HUBs to use them.

### CODE REVIEW

**Comment by Exxion** [comment]
1. Why the `spawn()`s?
2. Won't the portals make an audible sound when ghosts go through them?

**Comment by Exxion** [comment]
Hold on the hub one is already a feature. It already works in the game right now.

EDIT: They're actually *both* already features, but it appears that the portal one is bugged at the moment.

File: code/game/objects/effects/portals.dm [hunk 1]
```diff
@@ -161,7 +165,7 @@ var/list/portal_cache = list()
 /obj/effect/portal/proc/teleport(atom/movable/M as mob|obj)
 	if(istype(M, /obj/effect)) //sparks don't teleport
 		return
-	if (M.anchored && !istype(M, /obj/mecha) && !istype(M, /obj/item/projectile))
+	if (!isobserver(M) && M.anchored && !istype(M, /obj/mecha) && !istype(M, /obj/item/projectile))
```
Comment (Exxion) [inline code/game/objects/effects/portals.dm:164]: This is the only necessary change to allow ghosts to click through portals.

**Comment by Dacendeth** [comment]
We totally had this at one point but it was removed because it made portal sparks and noises.

**Comment by Exxion** [comment]
I looked back in time and I'm pretty sure it was actually removed by mistake

**Comment by DeityLink** [comment]
There I fixed it, and also made sure that ghosts don't produce sparks or sounds when moving through portals.

**Pull request merged**